website/src/components/Chat/ChatConversation.tsx

```diff
@@ -108,25 +117,115 @@ export const ChatConversation = memo(function ChatConversation({ chatId, getConf
+        chat_id: chatId,
+        parent_id: parentId,
+        model_config_name: model_config_name,
+        sampling_parameters: sampling_parameters,
+        plugins: plugins,
+      };
+
+      let draft_messages: InferenceMessage[];
+      try {
+        const promises = Array.from({ length: NUM_GENERATED_DRAFTS }, () =>
+          post(API_ROUTES.CREATE_ASSISTANT_MESSAGE, { arg: assistant_arg })
+        );
+
+        draft_messages = await Promise.all(promises);
+      } catch (err) {
+        if (err instanceof OasstError) {
+          toast({
+            title: err.message,
+            status: "error",
+          });
+        }
+        setIsSending.off();
+        return;
+      }
+
+      setStreamedDrafts(Array(NUM_GENERATED_DRAFTS).fill(""));
+      const complete_draft_messages = await Promise.all(
+        draft_messages.map(async (draft_message, index) => {
+          const { body, status } = await fetch(API_ROUTES.STREAM_CHAT_MESSAGE(chatId, draft_message.id));
+
+          let inference_message: InferenceMessage | null;
+          if (status === 204) {
+            inference_message = await get(API_ROUTES.GET_MESSAGE(chatId, draft_message.id));
+          } else {
+            inference_message = await handleChatEventStream({
+              stream: body!,
+              onError: console.error,
+              onPending:
+                draft_messages.every((message) => message.state === "pending") && index === 0 ? setQueueInfo : null,
+              onToken: async (text) => {
+                setQueueInfo(null);
+                setStreamedDrafts((drafts) => [...drafts.slice(0, index), text, ...drafts.slice(index + 1)]);
+                await new Promise(requestAnimationFrame);
+              },
+            });
+          }
+
+          setStreamedDrafts((drafts) => [
+            ...drafts.slice(0, index),
+            inference_message.content,
+            ...drafts.slice(index + 1),
+          ]);
+          await new Promise(requestAnimationFrame);
+
+          return inference_message;
+        })
+      );
+
+      setDraftMessages((draftMessages) => [...draftMessages, complete_draft_messages]);
+
+      setQueueInfo(null);
+      setIsSending.off();
+      setIsAwaitingMessageSelect.on();
+    },
+    [
+      getConfigValues,
+      setIsSending,
+      setStreamedDrafts,
+      setDraftMessages,
+      setQueueInfo,
+      setIsAwaitingMessageSelect,
+      NUM_GENERATED_DRAFTS,
+      toast,
+    ]
   );
 
   const { messagesEndRef, scrollableNodeProps, updateEnableAutoScroll, activateAutoScroll } = useAutoScroll(
     messages,
-    streamedResponse
+    streamedResponse,
+    draftMessages,
+    streamedDrafts
   );
 
   const sendPrompterMessage = useCallback(async () => {
     const content = inputRef.current?.value.trim();
     if (!content || isSending) {
       return;
     }
+
+    if (isAwaitingMessageSelect) {
+      return toast({
+        title: "Please select a draft to continue.",
```

Comment:
This string should be added to `public/locales/en/chat.json` for proper app localization.